appengine/flexible/TwelveFactor/Startup.cs

```diff
@@ -0,0 +1,172 @@
+﻿﻿/*
+ * Copyright (c) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+using Google.Cloud.Diagnostics.AspNetCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using TwelveFactor.Services;
+
+namespace TwelveFactor
+{
+    public class Startup
+    {
+        // Polling a file on Cloud Storage at the rate of
+        // once every 2 minutes will cost approximately $0.01 per month.
+        // See https://cloud.google.com/storage/pricing#network-pricing
+        CloudStorageFileProvider _cloudStorage = new CloudStorageFileProvider(
+            TimeSpan.FromMinutes(2));
+        public Startup(IHostingEnvironment env)
+        {
+            // Have to build the configuration twice.

```

Comment:
Ouch, is there no better way to handle this? Creating the configuration twice seems like a huge inconvenience. Maybe we have to find a way for the `Google.Api.Gax.Platform.Instance()` to return the right value when running locally?

My problem with this comes from the fact that we're doing this so the user can override the project id in the local `appsettings.{env}.json` file. We should find a better way of doing this. Could we simulate this with environment variables or something?